OmniOS build fails when building with autoconf/make: Incompatible defintion of `asctime_r`

The Gerbera OmniOS CI build fails when integrating the latest release (1.14.26) due to an incompatible definition of `asctime_r`.
```
../../upnp/src/genlib/net/http/webserver.c: In function 'web_server_asctime_r':
  ../../upnp/src/genlib/net/http/webserver.c:153:9: error: too many arguments to function 'asctime_r'
    153 |         asctime_r(tm, buf, ASCTIME_R_BUFFER_SIZE);
        |         ^~~~~~~~~
  In file included from /usr/include/sys/time.h:489,
                   from /usr/include/sys/select.h:53,
                   from /usr/include/sys/types.h:687,
                   from ../../upnp/src/threadutil/FreeList.h:43,
                   from ../../upnp/src/threadutil/LinkedList.h:39,
                   from ../../upnp/src/inc/httpparser.h:40,
                   from ../../upnp/src/inc/webserver.h:35,
                   from ../../upnp/src/genlib/net/http/webserver.c:44:
  /usr/include/time.h:248:14: note: declared here
    248 | extern char *asctime_r(const struct tm *_RESTRICT_KYWD, char *_RESTRICT_KYWD);
        |              ^~~~~~~~~
  make[3]: *** [Makefile:1476: src/genlib/net/http/libupnp_la-webserver.lo] Error 1
```
https://github.com/gerbera/gerbera/actions/runs/21766187357/job/62802490840?pr=3775

This commit https://github.com/pupnp/pupnp/commit/363eff0b28026cffff3525a9e40aa73932249432, added an OmniOS/Sun specific call to `asctime_r`. 
https://github.com/pupnp/pupnp/blob/76246bc7e35a83ae483fbbe73e34eedf05ba0105/upnp/src/genlib/net/http/webserver.c#L147-L155

but because autoconf/make defines `_POSIX_PTHREAD_SEMANTICS`, the OS is providing the POSIX-compliant 2 argument form of `asctime_r`.
https://github.com/pupnp/pupnp/blob/76246bc7e35a83ae483fbbe73e34eedf05ba0105/m4/ax_pthread.m4#L450-L458

I _suspect_ this wasn't caught in CI because the pupnp CI uses Cmake to build and this define is not present.

A quick fix might be
```diff
---	#elif defined(__sun__)
+++	#elif defined(__sun__) && !defined(_POSIX_PTHREAD_SEMANTICS)
static char *web_server_asctime_r(const struct tm *tm, char *buf)
{
	if (tm == NULL || buf == NULL)
		return NULL;

	asctime_r(tm, buf, ASCTIME_R_BUFFER_SIZE);
	return buf;
}
	#else
```

There is something weird in our OmniOS CI build. There are errors in `ctest`, but they don't abort the CI pipeline:

```bash
Run 'run' in VM
  run: cmake \
  -S . \
  -B build \
  -DCMAKE_BUILD_TYPE=Debug \
  -DBUILD_TESTING=ON \
  -DDOWNLOAD_AND_BUILD_DEPS=ON \
  -DUPNP_ENABLE_OPEN_SSL=ON \
  -DCMAKE_C_FLAGS="-Wall -Wpedantic" \
  -DCMAKE_INSTALL_PREFIX=/home/runner/work/pupnp/pupnp/test-install \
   -DUPNP_ENABLE_UNSPECIFIED_SERVER=ON -DUPNP_ENABLE_BLOCKING_TCP_CONNECTIONS=ON
  cmake --build build --config Debug
  cd build
  ctest --output-on-failure
  cd ..
  cmake -DBUILD_DIR=build -P cmake/post-test.cmake
  cmake --install build --config Debug
  
  exec ssh: cd $GITHUB_WORKSPACE;
  cmake \
  -S . \
  -B build \
  -DCMAKE_BUILD_TYPE=Debug \
  -DBUILD_TESTING=ON \
  -DDOWNLOAD_AND_BUILD_DEPS=ON \
  -DUPNP_ENABLE_OPEN_SSL=ON \
  -DCMAKE_C_FLAGS="-Wall -Wpedantic" \
  -DCMAKE_INSTALL_PREFIX=/home/runner/work/pupnp/pupnp/test-install \
   -DUPNP_ENABLE_UNSPECIFIED_SERVER=ON -DUPNP_ENABLE_BLOCKING_TCP_CONNECTIONS=ON
  cmake --build build --config Debug
  cd build
  ctest --output-on-failure
  cd ..
  cmake -DBUILD_DIR=build -P cmake/post-test.cmake
  cmake --install build --config Debug
  /usr/bin/bash /home/runner/work/_actions/vmactions/omnios-vm/v1/run.sh execSSH
  Config file: omnios-r151054.conf
  Pseudo-terminal will not be allocated because stdin is not a terminal.
  Last login: Fri Nov 14 03:30:17 2025 from 192.168.122.1
  -- Could NOT find Git (missing: GIT_EXECUTABLE) 
  Hint: The project() command has not yet been called.  It sets up system-specific search paths.
  -- Setting package-version to 1.14.26
  -- Setting ixml-soversion to 11.1.4
  -- Setting upnp-soversion to 17.2.6
  -- package-version is 1.14.26
  -- The C compiler identification is GNU 14.2.0
  -- The CXX compiler identification is GNU 14.2.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /usr/bin/gcc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /usr/bin/g++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Found OpenSSL: /usr/lib/64/libcrypto.so (found version "3.5.4")
  -- Looking for sys/socket.h
  -- Looking for sys/socket.h - found
  -- Looking for ws2tcpip.h
  -- Looking for ws2tcpip.h - not found
  -- Looking for sys/types.h
  -- Looking for sys/types.h - found
  -- Looking for stdint.h
  -- Looking for stdint.h - found
  -- Looking for stddef.h
  -- Looking for stddef.h - found
  -- Check size of socklen_t
  -- Check size of socklen_t - done
  -- Check size of off_t
  -- Check size of off_t - done
  -- System uses 64 bit, no flags needed
  -- Looking for fseeko
  -- Looking for fseeko - found
  -- Looking for strnlen
  -- Looking for strnlen - found
  -- Looking for strndup
  -- Looking for strndup - found
  -- Performing Test HAVE_MACRO_PREFIX_MAP
  -- Performing Test HAVE_MACRO_PREFIX_MAP - Success
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
  -- Found Threads: TRUE
  -- Check size of pthread_rwlock_t
  -- Check size of pthread_rwlock_t - done
  -- Could NOT find GTest (missing: GTest_DIR)
  -- Configuring done (3.6s)
  -- Generating done (0.1s)
  -- Build files have been written to: /home/runner/work/pupnp/pupnp/build
  [  0%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/attr.c.o
  [  1%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/document.c.o
  [  1%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/element.c.o
  [  2%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/ixml.c.o
  [  3%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/ixmldebug.c.o
  [  3%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/ixmlmembuf.c.o
  [  4%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/ixmlparser.c.o
  [  5%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/namedNodeMap.c.o
  [  5%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/node.c.o
  [  6%] Building C object ixml/CMakeFiles/ixml_shared.dir/src/nodeList.c.o
  [  7%] Linking C shared library libixml.so
  [  7%] Built target ixml_shared
  [  7%] Building C object ixml/CMakeFiles/ixml_static.dir/src/attr.c.o
  [  8%] Building C object ixml/CMakeFiles/ixml_static.dir/src/document.c.o
  [  8%] Building C object ixml/CMakeFiles/ixml_static.dir/src/element.c.o
  [  9%] Building C object ixml/CMakeFiles/ixml_static.dir/src/ixml.c.o
  [ 10%] Building C object ixml/CMakeFiles/ixml_static.dir/src/ixmldebug.c.o
  [ 10%] Building C object ixml/CMakeFiles/ixml_static.dir/src/ixmlmembuf.c.o
  [ 11%] Building C object ixml/CMakeFiles/ixml_static.dir/src/ixmlparser.c.o
  [ 12%] Building C object ixml/CMakeFiles/ixml_static.dir/src/namedNodeMap.c.o
  [ 12%] Building C object ixml/CMakeFiles/ixml_static.dir/src/node.c.o
  [ 13%] Building C object ixml/CMakeFiles/ixml_static.dir/src/nodeList.c.o
  [ 14%] Linking C static library libixml.a
  [ 14%] Built target ixml_static
  [ 15%] Building C object ixml/test/CMakeFiles/test-ixml-shared.dir/test_document.c.o
  [ 15%] Linking C executable test-ixml-shared
  [ 15%] Built target test-ixml-shared
  [ 16%] Building C object ixml/test/CMakeFiles/test-ixml-static.dir/test_document.c.o
  [ 17%] Linking C executable test-ixml-static
  [ 17%] Built target test-ixml-static
  [ 17%] Building C object ixml/test/CMakeFiles/test-ixml-poc-gh-506-shared.dir/poc_gh_506.c.o
  [ 18%] Linking C executable test-ixml-poc-gh-506-shared
  [ 18%] Built target test-ixml-poc-gh-506-shared
  [ 18%] Building C object ixml/test/CMakeFiles/test-ixml-poc-gh-506-static.dir/poc_gh_506.c.o
  [ 19%] Linking C executable test-ixml-poc-gh-506-static
  [ 19%] Built target test-ixml-poc-gh-506-static
  [ 20%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/upnpapi.c.o
  [ 20%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/upnpdebug.c.o
  [ 21%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpActionComplete.c.o
  [ 22%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpActionRequest.c.o
  [ 22%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpDiscovery.c.o
  [ 23%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpEvent.c.o
  [ 24%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpEventSubscribe.c.o
  [ 24%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpExtraHeaders.c.o
  [ 25%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpFileInfo.c.o
  [ 25%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpStateVarComplete.c.o
  [ 26%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpStateVarRequest.c.o
  [ 27%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpString.c.o
  [ 27%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/UpnpSubscriptionRequest.c.o
  [ 28%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/client_table/GenlibClientSubscription.c.o
  [ 29%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/client_table/client_table.c.o
  [ 29%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/miniserver/miniserver.c.o
  [ 30%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/net/sock.c.o
  [ 31%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/net/http/httpparser.c.o
  [ 31%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/net/http/httpreadwrite.c.o
  [ 32%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/net/http/parsetools.c.o
  [ 32%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/net/http/statcodes.c.o
  [ 33%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/net/http/webserver.c.o
  [ 34%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/net/uri/uri.c.o
  [ 34%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/service_table/service_table.c.o
  [ 35%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/util/list.c.o
  [ 36%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/util/membuffer.c.o
  [ 36%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/util/strintmap.c.o
  [ 37%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/genlib/util/util.c.o
  [ 38%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/threadutil/FreeList.c.o
  [ 38%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/threadutil/LinkedList.c.o
  [ 39%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/threadutil/ThreadPool.c.o
  [ 39%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/threadutil/TimerThread.c.o
  [ 40%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/urlconfig/urlconfig.c.o
  [ 41%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/gena/gena_device.c.o
  [ 41%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/gena/gena_ctrlpt.c.o
  [ 42%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/gena/gena_callback2.c.o
  [ 43%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/soap/soap_device.c.o
  [ 43%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/soap/soap_ctrlpt.c.o
  [ 44%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/soap/soap_common.c.o
  [ 45%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/ssdp/SSDPResultData.c.o
  [ 45%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/ssdp/SSDPResultDataCallback.c.o
  [ 46%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/ssdp/ssdp_device.c.o
  [ 46%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/ssdp/ssdp_ctrlpt.c.o
  [ 47%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/ssdp/ssdp_server.c.o
  [ 48%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/uuid/md5.c.o
  [ 48%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/uuid/sysdep.c.o
  [ 49%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/uuid/uuid.c.o
  [ 50%] Building C object upnp/CMakeFiles/upnp_shared.dir/src/api/upnptools.c.o
  [ 50%] Linking C shared library libupnp.so
  [ 50%] Built target upnp_shared
  [ 51%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/upnpapi.c.o
  [ 52%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/upnpdebug.c.o
  [ 52%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpActionComplete.c.o
  [ 53%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpActionRequest.c.o
  [ 53%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpDiscovery.c.o
  [ 54%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpEvent.c.o
  [ 55%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpEventSubscribe.c.o
  [ 55%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpExtraHeaders.c.o
  [ 56%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpFileInfo.c.o
  [ 57%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpStateVarComplete.c.o
  [ 57%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpStateVarRequest.c.o
  [ 58%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpString.c.o
  [ 59%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/UpnpSubscriptionRequest.c.o
  [ 59%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/client_table/GenlibClientSubscription.c.o
  [ 60%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/client_table/client_table.c.o
  [ 60%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/miniserver/miniserver.c.o
  [ 61%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/net/sock.c.o
  [ 62%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/net/http/httpparser.c.o
  [ 62%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/net/http/httpreadwrite.c.o
  [ 63%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/net/http/parsetools.c.o
  [ 64%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/net/http/statcodes.c.o
  [ 64%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/net/http/webserver.c.o
  [ 65%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/net/uri/uri.c.o
  [ 66%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/service_table/service_table.c.o
  [ 66%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/util/list.c.o
  [ 67%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/util/membuffer.c.o
  [ 67%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/util/strintmap.c.o
  [ 68%] Building C object upnp/CMakeFiles/upnp_static.dir/src/genlib/util/util.c.o
  [ 69%] Building C object upnp/CMakeFiles/upnp_static.dir/src/threadutil/FreeList.c.o
  [ 69%] Building C object upnp/CMakeFiles/upnp_static.dir/src/threadutil/LinkedList.c.o
  [ 70%] Building C object upnp/CMakeFiles/upnp_static.dir/src/threadutil/ThreadPool.c.o
  [ 71%] Building C object upnp/CMakeFiles/upnp_static.dir/src/threadutil/TimerThread.c.o
  [ 71%] Building C object upnp/CMakeFiles/upnp_static.dir/src/urlconfig/urlconfig.c.o
  [ 72%] Building C object upnp/CMakeFiles/upnp_static.dir/src/gena/gena_device.c.o
  [ 73%] Building C object upnp/CMakeFiles/upnp_static.dir/src/gena/gena_ctrlpt.c.o
  [ 73%] Building C object upnp/CMakeFiles/upnp_static.dir/src/gena/gena_callback2.c.o
  [ 74%] Building C object upnp/CMakeFiles/upnp_static.dir/src/soap/soap_device.c.o
  [ 74%] Building C object upnp/CMakeFiles/upnp_static.dir/src/soap/soap_ctrlpt.c.o
  [ 75%] Building C object upnp/CMakeFiles/upnp_static.dir/src/soap/soap_common.c.o
  [ 76%] Building C object upnp/CMakeFiles/upnp_static.dir/src/ssdp/SSDPResultData.c.o
  [ 76%] Building C object upnp/CMakeFiles/upnp_static.dir/src/ssdp/SSDPResultDataCallback.c.o
  [ 77%] Building C object upnp/CMakeFiles/upnp_static.dir/src/ssdp/ssdp_device.c.o
  [ 78%] Building C object upnp/CMakeFiles/upnp_static.dir/src/ssdp/ssdp_ctrlpt.c.o
  [ 78%] Building C object upnp/CMakeFiles/upnp_static.dir/src/ssdp/ssdp_server.c.o
  [ 79%] Building C object upnp/CMakeFiles/upnp_static.dir/src/uuid/md5.c.o
  [ 80%] Building C object upnp/CMakeFiles/upnp_static.dir/src/uuid/sysdep.c.o
  [ 80%] Building C object upnp/CMakeFiles/upnp_static.dir/src/uuid/uuid.c.o
  [ 81%] Building C object upnp/CMakeFiles/upnp_static.dir/src/api/upnptools.c.o
  [ 82%] Linking C static library libupnp.a
  [ 82%] Built target upnp_static
  [ 83%] Building C object upnp/sample/CMakeFiles/tv_ctrlpt.dir/common/sample_util.c.o
  [ 83%] Building C object upnp/sample/CMakeFiles/tv_ctrlpt.dir/common/tv_ctrlpt.c.o
  [ 84%] Building C object upnp/sample/CMakeFiles/tv_ctrlpt.dir/linux/tv_ctrlpt_main.c.o
  [ 85%] Linking C executable tv_ctrlpt
  [ 85%] Built target tv_ctrlpt
  [ 85%] Building C object upnp/sample/CMakeFiles/tv_combo.dir/common/sample_util.c.o
  [ 86%] Building C object upnp/sample/CMakeFiles/tv_combo.dir/common/tv_ctrlpt.c.o
  [ 87%] Building C object upnp/sample/CMakeFiles/tv_combo.dir/common/tv_device.c.o
  In file included from /home/runner/work/pupnp/pupnp/upnp/sample/common/sample_util.h:47,
                   from /home/runner/work/pupnp/pupnp/upnp/sample/common/tv_device.h:54,
                   from /home/runner/work/pupnp/pupnp/upnp/sample/common/tv_device.c:44:
  /home/runner/work/pupnp/pupnp/build/upnp/sample/common/config_sample.h:17:9: warning: "_LARGEFILE_SOURCE" redefined
     17 | #define _LARGEFILE_SOURCE 0
        |         ^~~~~~~~~~~~~~~~~
  In file included from /usr/include/signal.h:35,
                   from /home/runner/work/pupnp/pupnp/upnp/sample/common/tv_device.h:47:
  /usr/include/sys/feature_tests.h:239:9: note: this is the location of the previous definition
    239 | #define _LARGEFILE_SOURCE       1
        |         ^~~~~~~~~~~~~~~~~
  [ 87%] Building C object upnp/sample/CMakeFiles/tv_combo.dir/linux/tv_combo_main.c.o
  [ 88%] Linking C executable tv_combo
  [ 88%] Built target tv_combo
  [ 88%] Building C object upnp/sample/CMakeFiles/tv_device.dir/common/sample_util.c.o
  [ 89%] Building C object upnp/sample/CMakeFiles/tv_device.dir/common/tv_device.c.o
  In file included from /home/runner/work/pupnp/pupnp/upnp/sample/common/sample_util.h:47,
                   from /home/runner/work/pupnp/pupnp/upnp/sample/common/tv_device.h:54,
                   from /home/runner/work/pupnp/pupnp/upnp/sample/common/tv_device.c:44:
  /home/runner/work/pupnp/pupnp/build/upnp/sample/common/config_sample.h:17:9: warning: "_LARGEFILE_SOURCE" redefined
     17 | #define _LARGEFILE_SOURCE 0
        |         ^~~~~~~~~~~~~~~~~
  In file included from /usr/include/signal.h:35,
                   from /home/runner/work/pupnp/pupnp/upnp/sample/common/tv_device.h:47:
  /usr/include/sys/feature_tests.h:239:9: note: this is the location of the previous definition
    239 | #define _LARGEFILE_SOURCE       1
        |         ^~~~~~~~~~~~~~~~~
  [ 89%] Building C object upnp/sample/CMakeFiles/tv_device.dir/linux/tv_device_main.c.o
  [ 90%] Linking C executable tv_device
  [ 90%] Built target tv_device
  [ 91%] Building C object upnp/test/CMakeFiles/test-upnp-list.dir/test_list.c.o
  [ 91%] Linking C executable test-upnp-list
  [ 91%] Built target test-upnp-list
  [ 92%] Building C object upnp/test/CMakeFiles/test-upnp-list-static.dir/test_list.c.o
  [ 93%] Linking C executable test-upnp-list-static
  [ 93%] Built target test-upnp-list-static
  [ 93%] Building C object upnp/test/CMakeFiles/test-upnp-init.dir/test_init.c.o
  [ 94%] Linking C executable test-upnp-init
  [ 94%] Built target test-upnp-init
  [ 95%] Building C object upnp/test/CMakeFiles/test-upnp-init-static.dir/test_init.c.o
  [ 95%] Linking C executable test-upnp-init-static
  Undefined			first referenced
   symbol  			    in file
  bind                                ../libupnp.a(miniserver.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  recv                                ../libupnp.a(sock.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  send                                ../libupnp.a(sock.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  getsockname                         ../libupnp.a(miniserver.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  accept                              ../libupnp.a(miniserver.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  listen                              ../libupnp.a(miniserver.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  if_nametoindex                      ../libupnp.a(upnpapi.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  sendto                              ../libupnp.a(miniserver.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  socket                              ../libupnp.a(miniserver.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  __getifaddrs                        ../libupnp.a(upnpapi.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  setsockopt                          ../libupnp.a(miniserver.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  connect                             ../libupnp.a(httpreadwrite.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  in6addr_any                         ../libupnp.a(ssdp_server.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  getaddrinfo                         ../libupnp.a(uri.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  recvfrom                            ../libupnp.a(miniserver.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  freeaddrinfo                        ../libupnp.a(uri.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  gai_strerror                        ../libupnp.a(ssdp_device.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  freeifaddrs                         ../libupnp.a(upnpapi.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  shutdown                            ../libupnp.a(miniserver.c.o)  (symbol belongs to implicit dependency /lib/amd64/libsocket.so.1)
  ld: fatal: symbol referencing errors. No output written to test-upnp-init-static
  collect2: error: ld returned 1 exit status
  *** Error code 1
  The following command caused the error:
  cd /home/runner/work/pupnp/pupnp/build/upnp/test && /opt/ooce/bin/cmake -E cmake_link_script CMakeFiles/test-upnp-init-static.dir/link.txt --verbose=
  make: Fatal error: Command failed for target `upnp/test/test-upnp-init-static'
  Current working directory /home/runner/work/pupnp/pupnp/build
  *** Error code 1
  The following command caused the error:
  /usr/bin/make -s -f upnp/test/CMakeFiles/test-upnp-init-static.dir/build.make upnp/test/CMakeFiles/test-upnp-init-static.dir/build
  make: Fatal error: Command failed for target `upnp/test/CMakeFiles/test-upnp-init-static.dir/all'
  Current working directory /home/runner/work/pupnp/pupnp/build
  *** Error code 1
  The following command caused the error:
  /usr/bin/make -s -f CMakeFiles/Makefile2 all
  make: Fatal error: Command failed for target `all'
  Test project /home/runner/work/pupnp/pupnp/build
        Start  1: test-ixml
   1/12 Test  #1: test-ixml ........................   Passed    0.01 sec
        Start  2: test-ixml-static
   2/12 Test  #2: test-ixml-static .................   Passed    0.01 sec
        Start  3: test-ixml-poc-gh-506
   3/12 Test  #3: test-ixml-poc-gh-506 .............   Passed    0.00 sec
        Start  4: test-ixml-poc-gh-506-static
   4/12 Test  #4: test-ixml-poc-gh-506-static ......   Passed    0.00 sec
        Start  5: test-upnp-list
   5/12 Test  #5: test-upnp-list ...................   Passed    0.01 sec
        Start  6: test-upnp-list-static
   6/12 Test  #6: test-upnp-list-static ............   Passed    0.01 sec
        Start  7: test-upnp-init
   7/12 Test  #7: test-upnp-init ...................   Passed    0.01 sec
        Start  8: test-upnp-init-static
  Could not find executable /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-init-static
  Looked in the following places:
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-init-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-init-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-init-static
  Unable to find executable: /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-init-static
   8/12 Test  #8: test-upnp-init-static ............***Not Run   0.00 sec
        Start  9: test-upnp-log
  Could not find executable /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log
  Looked in the following places:
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-log
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-log
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-log
   9/12 Test  #9: test-upnp-log ....................***Not Run   0.00 sec
        Start 10: test-upnp-log-static
  Could not find executable /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log-static
  Looked in the following places:
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-log-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-log-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-log-static
  10/12 Test #10: test-upnp-log-static .............***Not Run   0.00 sec
        Start 11: test-upnp-url
  Could not find executable /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url
  Looked in the following places:
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-url
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-url
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-url
  11/12 Test #11: test-upnp-url ....................***Not Run   0.00 sec
        Start 12: test-upnp-url-static
  Could not find executable /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url-static
  Looked in the following places:
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-url-static
  /home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Release/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Debug/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/MinSizeRel/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/RelWithDebInfo/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Deployment/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-url-static
  home/runner/work/pupnp/pupnp/build/upnp/test/Development/test-upnp-url-static
  12/12 Test #12: test-upnp-url-static .............Unable to find executable: /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log
  Unable to find executable: /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-log-static
  Unable to find executable: /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url
  Unable to find executable: /home/runner/work/pupnp/pupnp/build/upnp/test/test-upnp-url-static
  ***Not Run   0.00 sec
  
  58% tests passed, 5 tests failed out of 12
  
  Total Test time (real) =   0.06 sec
  
  The following tests FAILED:
  	  8 - test-upnp-init-static (Not Run)
  	  9 - test-upnp-log (Not Run)
  	 10 - test-upnp-log-static (Not Run)
  	 11 - test-upnp-url (Not Run)
  	 12 - test-upnp-url-static (Not Run)
  Errors while running CTest
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:429]: Inside UpnpInitPreamble
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:566]: UpnpInit2 with IfName=NULL, DestPort=0.
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:4033]: Interface name=e1000g0, index=2, v4=192.168.122.243, v6=, ULA or GUA v6=
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:499]: Entering UpnpInitStartServers
  2025-11-14 03:32:22 UPNP-MSER-2: Thread:0x1 [.upnp/src/genlib/miniserver/miniserver.c:1207]: get_miniserver_sockets: resuseaddr is set.
  2025-11-14 03:32:22 UPNP-MSER-2: Thread:0x1 [.upnp/src/genlib/miniserver/miniserver.c:884]: sockfd = 5, .... port = 49152
  2025-11-14 03:32:22 UPNP-MSER-2: Thread:0x1 [.upnp/src/genlib/miniserver/miniserver.c:1231]: get_miniserver_sockets: bind successful
  2025-11-14 03:32:22 UPNP-MSER-2: Thread:0x1 [.upnp/src/genlib/miniserver/miniserver.c:884]: sockfd = 6, .... port = 57810
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:532]: Exiting UpnpInitStartServers
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:687]: Inside UpnpFinish: UpnpSdkInit is 1
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:694]: UpnpFinish: UpnpSdkInit is ONE
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:697]: Send Thread Pool
  High Jobs pending: 0
  Med Jobs Pending: 0
  Low Jobs Pending: 0
  Average wait in High Q in milliseconds: 0.000000
  Average wait in Med Q in milliseconds: 0.000000
  Average wait in Low Q in milliseconds: 0.000000
  Max Threads Used: 3
  Worker Threads: 0
  Persistent Threads: 1
  Idle Threads: 2
  Total Threads: 3
  Total Work Time: 0.000000
  Total Idle Time: 0.000000
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:699]: Recv Thread Pool
  High Jobs pending: 0
  Med Jobs Pending: 0
  Low Jobs Pending: 0
  Average wait in High Q in milliseconds: 0.000000
  Average wait in Med Q in milliseconds: 0.000000
  Average wait in Low Q in milliseconds: 0.000000
  Max Threads Used: 2
  Worker Threads: 0
  Persistent Threads: 0
  Idle Threads: 2
  Total Threads: 2
  Total Work Time: 0.000000
  Total Idle Time: 0.000000
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:702]: MiniServer Thread Pool
  High Jobs pending: 0
  Med Jobs Pending: 0
  Low Jobs Pending: 0
  Average wait in High Q in milliseconds: 0.000000
  Average wait in Med Q in milliseconds: 0.000000
  Average wait in Low Q in milliseconds: 0.000000
  Max Threads Used: 3
  Worker Threads: 0
  Persistent Threads: 1
  Idle Threads: 2
  Total Threads: 3
  Total Work Time: 0.000000
  Total Idle Time: 0.000000
  2025-11-14 03:32:22 UPNP-MSER-2: Thread:0x9 [.upnp/src/genlib/miniserver/miniserver.c:728]: Received response: ShutDown From host 127.0.0.1 
  2025-11-14 03:32:22 UPNP-MSER-1: Thread:0x9 [.upnp/src/genlib/miniserver/miniserver.c:735]: Received multicast packet: 
   ShutDown
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:729]: MiniServer Thread Pool
  High Jobs pending: 0
  Med Jobs Pending: 0
  Low Jobs Pending: 0
  Average wait in High Q in milliseconds: 0.000000
  Average wait in Med Q in milliseconds: 0.000000
  Average wait in Low Q in milliseconds: 0.000000
  Max Threads Used: 3
  Worker Threads: 0
  Persistent Threads: 0
  Idle Threads: 0
  Total Threads: 0
  Total Work Time: 0.000000
  Total Idle Time: 0.000000
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:733]: Send Thread Pool
  High Jobs pending: 0
  Med Jobs Pending: 0
  Low Jobs Pending: 0
  Average wait in High Q in milliseconds: 0.000000
  Average wait in Med Q in milliseconds: 0.000000
  Average wait in Low Q in milliseconds: 0.000000
  Max Threads Used: 3
  Worker Threads: 0
  Persistent Threads: 0
  Idle Threads: 3
  Total Threads: 3
  Total Work Time: 0.000000
  Total Idle Time: 0.000000
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:736]: Recv Thread Pool
  High Jobs pending: 0
  Med Jobs Pending: 0
  Low Jobs Pending: 0
  Average wait in High Q in milliseconds: 0.000000
  Average wait in Med Q in milliseconds: 0.000000
  Average wait in Low Q in milliseconds: 0.000000
  Max Threads Used: 2
  Worker Threads: 0
  Persistent Threads: 0
  Idle Threads: 0
  Total Threads: 0
  Total Work Time: 0.000000
  Total Idle Time: 0.000000
  2025-11-14 03:32:22 UPNP-API_-2: Thread:0x1 [.upnp/src/api/upnpapi.c:748]: Exiting UpnpFinish: UpnpSdkInit is :0:
  ```

TL;DR:
```bash
  The following tests FAILED:
  	  8 - test-upnp-init-static (Not Run)
  	  9 - test-upnp-log (Not Run)
  	 10 - test-upnp-log-static (Not Run)
  	 11 - test-upnp-url (Not Run)
  	 12 - test-upnp-url-static (Not Run)
  Errors while running CTest
```

Anyone know why the workflow did not abort? `ctest` returns an error status when the tests fails.

--- a/upnp/src/genlib/net/http/webserver.c
+++ b/upnp/src/genlib/net/http/webserver.c
@@ -144,7 +144,7 @@ static char *web_server_asctime_r(const struct tm *tm, char *buf)
 	asctime_s(buf, ASCTIME_R_BUFFER_SIZE, tm);
 	return buf;
 }
-	#elif defined(__sun__)
+	#elif defined(__sun__) && !defined(_POSIX_PTHREAD_SEMANTICS)
 static char *web_server_asctime_r(const struct tm *tm, char *buf)
 {
 	if (tm == NULL || buf == NULL)

